File: src/publisher_subscriber/src/py/ps_listener.py
#!/usr/bin/env python
import rospy
from rospy.exceptions import ROSInterruptException

# Importing String message from std_msgs package
from std_msgs.msg import String

# Callback function to handle new message
def callback(message):
	# To know the structure of message object, run
	# $ rosmsg info std_msgs/String
	rospy.loginfo(message.data)

# Define the actions of the node
def node():
	# Initialize the node
	# - Name of the node
	# - Whether to make the node different from each other
	# - If anonymous is true then a random number is appended to the name of the node for every
	#   instance of the node created using the rosrun command
	rospy.init_node("ps_listener", anonymous=False)

	# Create a subscriber
	# - Name of the topic
	# - Type of the topic
	# - Callback function to execute when a message arrives
	sub = rospy.Subscriber('/my_message', String, callback)

	# Start listening
	rospy.spin()


# Start the node when the file is executed
if __name__ == "__main__":
	try:
		node()
	except ROSInterruptException:
		pass

File: src/publisher_subscriber/src/cpp/ps_listener.cpp
#include <ros/ros.h>

// Importing String message from std_msgs package
#include "std_msgs/String.h"

// Callback function to handle new message
void callback(const std_msgs::String::ConstPtr& msg) {
	// To know the structure of message object, run
	// $ rosmsg info std_msgs/String
  ROS_INFO("%s", msg->data.c_str());
}

// Define actions of the node
int main(int argc, char* argv[]) {
  // Initialize the node
  // - argc and argv
  // - Name of the node
  ros::init(argc, argv, "ps_listener");

  // Create a Subscriber
  // - NodeHandle node
  // - Name of the topic
  // - Buffer size to hold messages
  // - Callback function to execute when a message arrives
  ros::NodeHandle node;
  ros::Subscriber sub = node.subscribe("/my_message", 10, callback);

  // Start listening
  ros::spin();

  return 0;
}
